### PR TITLE
Implement `RustcInternal` for `StableBody`

### PR DESCRIPTION
**Pull request by artemagvanian**
StableMIR has a way to convert internal `Body` into its stable counterpart, but not the other way around. Being able to perform the conversion both ways could allow using tools like `rustc_mir_dataflow`, which do not understand stable `Body`, from StableMIR.

r? celinval

### CODE REVIEW

File: compiler/rustc_smir/src/rustc_internal/internal.rs
Comment (celinval) [inline compiler/rustc_smir/src/rustc_internal/internal.rs:1]: can you please move this to its own module `internal/mir.rs`?

Comment (artemagvanian) [reply compiler/rustc_smir/src/rustc_internal/internal.rs:1]: I pulled all impls for MIR items into its own file (some of them were already implemented before me)

**Comment by oli-obk** [comment]
I don't think we should be doing this PR. The existing back-conversions are fairly minimal for the core types like `Ty`, `Const` and `Lifetime`. 

Instead things like data flow can be implemented out of tree on top of the stable MIR APIs.

**Comment by artemagvanian** [comment]
Thanks for your feedback, @oli-obk.

We implemented these conversions for our immediate need and wanted to see whether it would make sense to contribute them back to StableMIR. 

I agree that implementing data flow analysis infrastructure on top of StableMIR is better, and this is something we would like to do in the future.

**Pull request closed without merging**